SummerSchool/app/src/main/java/nl/rug/www/rugsummerschools/controller/MainActivity.java

```diff
@@ -0,0 +1,386 @@
+
+        FirebaseUser user = FirebaseAuth.getInstance().getCurrentUser();
+        if(user != null){
+            ArrayList<String> mlogInData = new ArrayList<>();
+            mlogInData.add(user.getPhotoUrl().toString());
+            mlogInData.add(user.getDisplayName());
+            mlogInData.add(user.getEmail());
+            mlogInData.add(user.getUid());
+            ContentsLab.get().setmLogInData(mlogInData);
+        }
+
+        mViewPager = (ViewPager)findViewById(R.id.main_view_pager);
+        mViewPager.addOnPageChangeListener(mSimpleOnPageChangeListener);
+        mViewPager.setAdapter(mFragmentStatePagerAdapter);
+    }
+
+    @SuppressLint("RestrictedApi")
+    private void disableShiftingNavigationMode() {
+        BottomNavigationMenuView menuView = (BottomNavigationMenuView) mNavigation.getChildAt(0);
+        try {
+            Field shiftingMode = menuView.getClass().getDeclaredField("mShiftingMode");
+            shiftingMode.setAccessible(true);
+            shiftingMode.setBoolean(menuView, false);
+            shiftingMode.setAccessible(false);
+            for(int i = 0; i < menuView.getChildCount(); i++) {
+                BottomNavigationItemView itemView = (BottomNavigationItemView) menuView.getChildAt(i);
+                itemView.setShiftingMode(false);
+                itemView.setChecked(itemView.getItemData().isChecked());
+            }
+        } catch (NoSuchFieldException | IllegalAccessException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void handleFacebookAccessToken(AccessToken token) {
+        Log.d(TAG, "handleFacebookAccessToken:" + token);
+
+        AuthCredential credential = FacebookAuthProvider.getCredential(token.getToken());
+        mAuth.signInWithCredential(credential)
+                .addOnCompleteListener(this, new OnCompleteListener<AuthResult>() {
+                    @Override
+                    public void onComplete(@NonNull Task<AuthResult> task) {
+                        if (task.isSuccessful()) {
+                            // Sign in success, update UI with the signed-in user's information
+                            Log.d(TAG, "signInWithCredential:success");
+                            FirebaseUser user = mAuth.getCurrentUser();
+                            updateUI(user);
+                        } else {
+                            // If sign in fails, display a message to the user.
+                            Log.w(TAG, "signInWithCredential:failure", task.getException());
+                            Toast.makeText(MainActivity.this, "Authentication failed.",
+                                    Toast.LENGTH_SHORT).show();
+                            updateUI(null);
+                        }
+
+                        // ...
+                    }
+                });
+    }
+
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        // Check if user is signed in (non-null) and update UI accordingly.
```

Comment:
there's no check in this method. is this an out of date comment?

Reply:
Yeah, it is kinda out dated one.